napari/_qt/widgets/_tests/test_qt_dims.py

```diff
@@ -15,7 +15,7 @@ def test_creating_view(qtbot):
     Test creating dims view.
     """
     ndim = 4
-    dims = Dims(ndim)
+    dims = Dims(ndim=ndim)
```

Comment:
pydantic models are kwarg only so `Dims(ndim)` no longer works, needs to be `Dims(ndim=ndim)`

Reply:
I saw that ... that's kind of a bummer, no more `Image(np.random.rand(10, 10))`?  might need to patch that...

Reply:
turns out this works for positional args:

```python
from pydantic import BaseModel

class TakesArgs(BaseModel):
    x: int
    y: int=4

    def __init__(self, x, **data):
        super().__init__(x=x, **data)
```

Reply:
Ok nice that's great! I think allowing `Colormaps(color)` is nice as that is public facing. I'm fine with `Dims(ndim=2)` though as that is highly internal. Will see how others feel during review

Reply:
is this PR just waiting for review?  Were you going to modify any of these `__init__` methods to allow position arguments in some cases?

Reply:
yup, waiting for review/ any one else to weigh in with a strong opinion on which of the `__ini__` to modify or not - I'm leaning to not modifying `Dims` as it is so internal, but modifying `Coloromap` as someone else might want to use it. If you like that then I'll do it now

Reply:
I have now done this